samples/publish-confirm/index.jsx

```diff
@@ -0,0 +1,195 @@
+import "babel-polyfill"
+import { init } from "contentful-ui-extensions-sdk"
+import React from "react"
+import ReactDOM from "react-dom"
+import { default as relativeDate } from "relative-date"
+import "@contentful/forma-36-react-components/dist/styles.css"
+import "@contentful/forma-36-fcss/dist/styles.css"
+import {
+  Button,
+  TextLink,
+  Paragraph,
+  Subheading
+} from "@contentful/forma-36-react-components"
+
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+    this.onClickPublish = this.onClickPublish.bind(this)
+    this.onClickUnpublish = this.onClickUnpublish.bind(this)
+    this.onUpdate = this.onUpdate.bind(this)
+    this.onError = this.onError.bind(this)
+
+    // Update component state when a field value changes
+    const fields = this.props.extension.entry.fields
+    for (let key in fields) {
+      fields[key].onValueChanged(this.onUpdate)
+    }
+
+    // Trigger update method to setup initial state
+    this.state = this.constructState()
+  }
+
+  constructState() {
+    const sys = this.getEntrySys()
```

Comment:
`this.getEntrySys()` is used only this this function, so I would inline the logic to make it easier to follow.